Celeste.Mod.mm/Mod/Everest/Everest.Updater.cs

```diff
@@ -232,27 +229,47 @@ public class Source {
                         name = name.Substring(0, indexOfBranch);
                     }
 
-                    return new Entry(name, branch, url, int.Parse(Regex.Match(split[1], @"\d+").Value));
+                    return new Entry(name, url, int.Parse(Regex.Match(split[1], @"\d+").Value), source);
                 };
 
-            private static Func<string, List<Entry>> AzureDataParser(string artifactFormat, int offset)
-                => (dataRaw) => {
+            public static Func<Source, string, List<Entry>> AzureBuildsParser(string artifactFormat, int offset)
+                => (source, dataRaw) => {
                     List<Entry> entries = new List<Entry>();
 
                     JObject root = JObject.Parse(dataRaw);
                     JArray list = root["value"] as JArray;
                     foreach (JObject build in list) {
-                        if (build["status"].ToObject<string>() != "completed" || build["result"].ToObject<string>() != "succeeded")
-                            continue;
+                        int id = build["id"].ToObject<int>();
+                        string url = string.Format(artifactFormat, id);
+                        Entry entry = new Entry((id + offset).ToString(), url, id + offset, source);
+                        try { entry.Description = build.SelectToken("triggerInfo['ci.message']").ToString().Split('\n')[0]; } catch { }
+                        entries.Add(entry);
+                    }
+
+                    return entries;
+                };
 
-                        string reason = build["reason"].ToObject<string>();
-                        if (reason != "manual" && reason != "individualCI")
+            public static Func<Source, string, List<Entry>> GitHubReleasesParser(int offset, bool prerelease = false)
+                => (source, dataRaw) => {
+                    List<Entry> entries = new List<Entry>();
+
+                    JArray list = JArray.Parse(dataRaw);
+                    foreach (JObject release in list) {
+                        if (prerelease != release["prerelease"].ToObject<bool>())
                             continue;
 
-                        int id = build["id"].ToObject<int>();
-                        string branch = build["sourceBranch"].ToObject<string>().Replace("refs/heads/", "");
-                        string url = string.Format(artifactFormat, id);
-                        entries.Add(new Entry((id + offset).ToString(), branch, url, id + offset));
+                        string build = Regex.Match(release["name"].ToString(), @"\d+$").Value;
+                        string url = null;
+                        foreach (JObject asset in release["assets"] as JArray) {
+                            if (asset["name"].ToString() == "main.zip") {
+                                url = asset["url"].ToString();
```

Comment:
should use `browser_download_url` instead of `url`

Reply:
This has been changed, although the GitHub Release Parser is no longer used for the main Everest branches.